summoner-cli/src/Summoner/Template/Nix.hs

```diff
@@ -0,0 +1,131 @@
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE QuasiQuotes    #-}
+
+module Summoner.Template.Nix
+       ( nixFiles
+       ) where
+
+import NeatInterpolation (text)
+
+import Summoner.Default (defaultNixCompiler, defaultNixPkgSet)
+import Summoner.GhcVer (nixCompiler)
+import Summoner.Settings (NixPkgSet (..), Settings (..))
+import Summoner.Tree (TreeFs (..))
+
+import qualified Data.Text as T
+
+nixFiles :: Settings -> [TreeFs]
+nixFiles Settings{..} =
+    [ shellNix
+    , releaseNix
+    , defaultNix
+    , nixDir
+    ]
+    where
+        defaultNix = File "default.nix" defaultNixT
+        shellNix = File "shell.nix" shellNixT
+        releaseNix = File "release.nix" releaseNixT
+
+        nixDir = Dir "nix" [fetchNixpkgsNix, nixpkgsNix, overridesNix]
+        fetchNixpkgsNix = File "fetchNixpkgs.nix" fetchNixpkgsNixT
+        nixpkgsNix = File "nixpkgs.nix" nixpkgsNixT
+        overridesNix = File "overrides.nix" overridesNixT
+
+        defaultNixT :: Text
+        defaultNixT =
+            [text|
+            { compiler ? "${defaultNixCompiler}"
+            , nixpkgs ? (import ./nix/nixpkgs.nix { inherit compiler; })
+            }:
+
+            with rec {
+              drv = nixpkgs.haskellPackages.${settingsRepo};
+            };
+
+            drv
+            |]
+
+        shellNixT :: Text
+        shellNixT =
+            [text|
+            (import ./default.nix {}).env
+            |]
+
+        releaseNixCompilers :: Text
+        releaseNixCompilers = T.unlines
+            ( map
+                  (\ghcV -> let compiler = nixCompiler ghcV
+                            in "  " <> compiler <> "  = import ./default.nix { compiler = \"" <> compiler <> "\";  };"
+                  )
+                  settingsTestedVersions
+            )
+
+        releaseNixT :: Text
+        releaseNixT =
+            [text|
+            {
+            $releaseNixCompilers
+            }
+            |]
+ 
+        fetchNixpkgsNixT :: Text
+        fetchNixpkgsNixT = T.unlines
+            [ "{ owner  # The owner of the nixpkgs"
+            , ", repo   # The repo of the nixpkgs"
+            , ", rev    # The Git revision of nixpkgs to fetch"
+            , ", sha256 # The SHA256 hash of the unpacked archive"
+            , "}:"
+            , "builtins.fetchTarball {"
+            , "  url = \"https://github.com/${owner}/${repo}/archive/${rev}.tar.gz\";"
```

Comment:
Is this correct? This is just text, so these fields will not be interpolated. If you'd like to use this as variables you should use `text` quasi-quoter here.